osconfig_tests/test_suites/patch/patch.go

```diff
@@ -0,0 +1,226 @@
+package patch
+
+import (
+	"context"
+	"fmt"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/osconfig_server"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/utils"
+	"github.com/golang/protobuf/ptypes/duration"
+	"github.com/kylelemons/godebug/pretty"
+	"log"
+	"path"
+	"regexp"
+	"sync"
+	"time"
+
+	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/compute"
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/junitxml"
+	api "google.golang.org/api/compute/v1"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/osconfig_tests/test_config"
+)
+
+const (
+	testSuiteName = "PatchTests"
+)
+
+var (
+	dump = &pretty.Config{IncludeUnexported: true}
+)
+
+var vf = func(inst *compute.Instance, vfString string, port int64, interval, timeout time.Duration) error {
+	return inst.WaitForSerialOutput(vfString, port, interval, timeout)
+}
+
+type patchTestSetup struct {
+	image         string
+	name          string
+	startup       *api.MetadataItems
+	vstring       string
+	assertTimeout time.Duration
+	vf            func(*compute.Instance, string, int64, time.Duration, time.Duration) error
+}
+
+func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junitxml.TestSuite, logger *log.Logger, testSuiteRegex, testCaseRegex *regexp.Regexp, testProjectConfig *testconfig.Project) {
+	defer tswg.Done()
+
+	if testSuiteRegex != nil && !testSuiteRegex.MatchString(testSuiteName) {
+		return
+	}
+
+	testSuite := junitxml.NewTestSuite(testSuiteName)
+	defer testSuite.Finish(testSuites)
+
+	logger.Printf("Running TestSuite %q", testSuite.Name)
+
+	suffix := utils.RandString(5)
+	image := "projects/debian-cloud/global/images/family/debian-9"
+
+	testSetup := []*patchTestSetup{
+		// Windows images.
+		&patchTestSetup{
+			image:   image,
+			name:    fmt.Sprintf("patch-test-%s-%s", path.Base(image), suffix),
```

Comment:
Is this the name that shows up in the dashboard? If so, `PatchTests-%s-%s` would be more consistent.

Reply:
this is the name on the dashboard 
https://github.com/GoogleCloudPlatform/compute-image-tools/pull/752/files#diff-f2be2b42126f00bd4171c9cd270b9f33R212. 

this gets converted to 
fmt.Sprintf("[%s] %s", classname, name)
which in our case would be "[PatchTests] [Imagename/testname] test description"

i understand your concern around consistency of test name. i am thinking of addressing the naming pattern in a separate PR, since all the 3 testsuites currently follow different patterns. Even worst is that in package manager, i have a very terrible logic that depends on the test name pattern. so it would be better if i take it separately.